pallets/settlement/src/lib.rs

```diff
@@ -1037,77 +1081,73 @@ impl<T: Trait> Module<T> {
                 SettlementDID.as_id(),
                 instruction_id,
             ));
-            result = DispatchResult::Err(Error::<T>::InstructionFailed.into());
-        } else {
-            // Verify that the venue still has the required permissions for the tokens involved.
-            let tickers: BTreeSet<Ticker> = legs.iter().map(|leg| leg.1.asset).collect();
-            let venue_id = Self::instruction_details(instruction_id).venue_id;
-            for ticker in &tickers {
-                if Self::venue_filtering(ticker) && !Self::venue_allow_list(ticker, venue_id) {
-                    Self::deposit_event(RawEvent::VenueUnauthorized(
-                        SettlementDID.as_id(),
-                        *ticker,
-                        venue_id,
-                    ));
-                    result = DispatchResult::Err(Error::<T>::UnauthorizedVenue.into());
-                }
+            Self::prune_instruction(instruction_id);
+            return Err(Error::<T>::InstructionFailed.into());
```

Comment:
Suppose we do `execute_instruction_retryable(id)` and we reach here. We return an error, which results in `execute_instruction_retryable` marking the status as failed, allowing `reschedule_instruction` to result in a future `execute_instruction`. However, the instruction has been pruned, so... Huston we have a problem?

Reply:
There's a contains key check tp prevent this